libr/debug/debug.c

```diff
@@ -661,13 +660,47 @@ R_API const char *r_debug_reason_to_string(int type) {
 	case R_DEBUG_REASON_TRAP: return "trap";
 	case R_DEBUG_REASON_SWI: return "software-interrupt";
 	case R_DEBUG_REASON_INT: return "interrupt";
-	case R_DEBUG_REASON_FPU: return "fpu";
 	case R_DEBUG_REASON_STEP: return "step";
 	case R_DEBUG_REASON_USERSUSP: return "suspended-by-user";
 	}
 	return "unhandled";
 }
 
+R_API const char *r_signal_to_human(int signum) {
+	switch (signum) {
+		case R_SIGNAL_SIGALRM: return "Alarm clock";
+		case R_SIGNAL_SIGBUS: return "Bus error";
+		case R_SIGNAL_SIGCHLD: return "";
+		case R_SIGNAL_SIGCONT: return "";
+	#ifdef __sparc
+		case R_SIGNAL_SIGEMT: return "hardware-error-signal";
+	#endif
+		case R_SIGNAL_SIGFPE: return "Floating point exception";
+		case R_SIGNAL_SIGHUP: return "Hangup";
+		case R_SIGNAL_SIGILL: return "Illegal instruction";
+		case R_SIGNAL_SIGKILL: return "Killed";
+		case R_SIGNAL_SIGPROF: return "Profiling timer expired";
+		case R_SIGNAL_SIGPWR: return "Power failure";
+		case R_SIGNAL_SIGQUIT: return "Quit";
+		case R_SIGNAL_SIGSEGV: return "Segmentation fault";
+		case R_SIGNAL_SIGSTKFLT: return "Stack fault";
+		case R_SIGNAL_SIGSTOP: return "Stopped";
+		case R_SIGNAL_SIGSYS: return "Bad system call";
+		case R_SIGNAL_SIGTERM: return "Terminated";
+		case R_SIGNAL_SIGTSTP: return "Stopped";
+		case R_SIGNAL_SIGTTIN: return "Stopped";
+		case R_SIGNAL_SIGTTOU: return "Stopped";
+		case R_SIGNAL_SIGURG: return "";
+		case R_SIGNAL_SIGUSR1: return "User defined signal 1";
+		case R_SIGNAL_SIGUSR2: return "User defined signal 2";
+		case R_SIGNAL_SIGVTALRM: return "Virtual timer expired";
+		case R_SIGNAL_SIGWINCH: return "";
+		case R_SIGNAL_SIGXCPU: return "CPU time limit exceeded";
+		case R_SIGNAL_SIGXFSZ: return "File size limit exceeded";
+		default: return "unhandled";
```

Comment:
We can return to tostring here. What do u think?

Reply:
There is already a such function in libr/util/signal.c I do not want to reimplement it.

I could move it from libr/util/signal.c to libr/debug/debug.c.